### PR TITLE
First refactor effort on march to first release version 0.1

### LINKED ISSUE #12: Error message for missing control_barcodes in 1.process-spots
In 1.process-spots at this point:
```
    # Number of non-targetting controls
    num_nt = passed_gene_df.query(
        f"{gene_cols[0]} in @control_barcodes"
    ).Cell_Count_Per_Gene.values[0]
```
it will error if there are no non-targeting controls at a site. This can happen because 1) biologically, the site actually doesn't contain control_barcodes, or 2) the barcodes you have listed in your config file don't match what is used in the experiment for control_barcodes.  
  
Add in an error message to explain this, ideally parsing between the conditions.

### LINKED ISSUE #26: process_cells config mismatch
In the config for process_cells we have `prefilter-features` as a Bool but in 2.process-cells.py is requires running 0.preprocess-sites.py first so that you have a prefilter file as input.

@gwaygenomics What is the goal of this config Bool? 
Do we want to update 2.process-cells.py to allow for not using a prefilter file? Or do we want to remove the input from the config?

### LINKED ISSUE #32: Metadata in 7.visualize-cell-summary.py
in [these lines](https://github.com/broadinstitute/pooled-cell-painting-profiling-recipe/blob/master/0.preprocess-sites/7.visualize-cell-summary.py#L89-L95) we load metadata and append to a `metadata_list`. However, this is never used in this script.

I did not see this when I reviewed #18, but I ran into it now when I came across this error:

```python
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
<ipython-input-8-235105add032> in <module>
     10     metadata_df = (
     11         pd.read_csv(metadata_file, sep="\t")
---> 12         .loc[:, metadata_col_list]
     13         .reset_index(drop=True)
     14     )

~/miniconda3/envs/pooled-cp/lib/python3.7/site-packages/pandas/core/indexing.py in __getitem__(self, key)
   1759                 except (KeyError, IndexError, AttributeError):
   1760                     pass
-> 1761             return self._getitem_tuple(key)
   1762         else:
   1763             # we by definition only have the 0th axis

~/miniconda3/envs/pooled-cp/lib/python3.7/site-packages/pandas/core/indexing.py in _getitem_tuple(self, tup)
   1286                 continue
   1287 
-> 1288             retval = getattr(retval, self.name)._getitem_axis(key, axis=i)
   1289 
   1290         return retval

~/miniconda3/envs/pooled-cp/lib/python3.7/site-packages/pandas/core/indexing.py in _getitem_axis(self, key, axis)
   1951                     raise ValueError("Cannot index with multidimensional key")
   1952 
-> 1953                 return self._getitem_iterable(key, axis=axis)
   1954 
   1955             # nested tuple slicing

~/miniconda3/envs/pooled-cp/lib/python3.7/site-packages/pandas/core/indexing.py in _getitem_iterable(self, key, axis)
   1592         else:
   1593             # A collection of keys
-> 1594             keyarr, indexer = self._get_listlike_indexer(key, axis, raise_missing=False)
   1595             return self.obj._reindex_with_indexers(
   1596                 {axis: [keyarr, indexer]}, copy=True, allow_dups=True

~/miniconda3/envs/pooled-cp/lib/python3.7/site-packages/pandas/core/indexing.py in _get_listlike_indexer(self, key, axis, raise_missing)
   1550 
   1551         self._validate_read_indexer(
-> 1552             keyarr, indexer, o._get_axis_number(axis), raise_missing=raise_missing
   1553         )
   1554         return keyarr, indexer

~/miniconda3/envs/pooled-cp/lib/python3.7/site-packages/pandas/core/indexing.py in _validate_read_indexer(self, key, indexer, axis, raise_missing)
   1652             if not (ax.is_categorical() or ax.is_interval()):
   1653                 raise KeyError(
-> 1654                     "Passing list-likes to .loc or [] with any missing labels "
   1655                     "is no longer supported, see "
   1656                     "https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#deprecate-loc-reindex-listlike"  # noqa:E501

KeyError: 'Passing list-likes to .loc or [] with any missing labels is no longer supported, see https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#deprecate-loc-reindex-listlike'
```

@ErinWeisbart - is it safe to remove metadata references in this script?

### PR DESCRIPTION
**Pull request by gwaygenomics**
This PR _mostly_ touches the preprocessing module. The PR also introduces minor changes in two files in the generate profiles module: `1.generate-profiles/profiling_config.yaml` and `1.generate-profiles/0.merge-single-cells.py`

## Changelog

* Abstracting command arguments (loading config file and `--force` flag) to new function `parse_command_args()`
* Updating "force" logic. There are now two ways to toggle overwriting (perhaps more decisions need to be made here, see #24)
  * This logic is encapsulated in a new function `check_if_write()`.
* [Updating language from `master` to `main` config](https://www.zdnet.com/article/github-to-replace-master-with-alternative-term-to-avoid-slavery-references/#:~:text=ZDNet%20Academy-,GitHub%20to%20replace%20%22master%22%20with%20alternative%20term%20to%20avoid%20slavery,its%20CEO%20said%20on%20Friday.).
* [BUG FIX] in `2.process-cells.py` we were renaming columns and assigning a `site` column without actually saving it to the dataframe ([here](https://github.com/broadinstitute/pooled-cell-painting-profiling-recipe/compare/master...gwaygenomics:refactor-to-first-release?expand=1#diff-60750dd97846eb5d02654ce784c9eb1fR173-R177))
* [BUG FIX] closes #32
* close #26
* [BUG FIX] closes #12
* Update script order

### CODE REVIEW

File: 0.preprocess-sites/README.md [hunk 1]
```diff
@@ -30,5 +30,5 @@ This step compiles all the results from the spot and cell processing and outputs
 ## Step 4 - Image and Segmentation Quality Control
 
 This step outputs a series of QC metrics pertaining to the actual image acquisition and image segmentation quality.
-The QC metrics are visualized by site across wells per experiment. 
+The QC metrics are visualized by site across wells per experiment.
 It also outputs lists of sites that trigger QC flags for confluence, image saturation, and image alignment.
```
Comment (ErinWeisbart) [inline 0.preprocess-sites/README.md:34]: Is your intent for this README to have each sentence on its own line (as we had discussed for the site_processing_config_README in the template)? If so, .md requires two spaces at the end of each line to make a line break.

Comment (gwaygenomics) [reply 0.preprocess-sites/README.md:34]: oh! Sorry - "one-sentence-per-line refers" to markdown style exclusively. My intent is for them to appear in paragraph form in the rendered markdown, but that in the raw markdown each sentence belongs on its own line.

**Comment by gwaygenomics** [comment]
Thanks for the review - sorry it took me so long to revisit

> Readme formatting

See my clarification above.

> Do we include both command line and config force_overwrite? (Discussed here)

Yep, we do. See https://github.com/broadinstitute/pooled-cell-painting-profiling-recipe/pull/34/files#diff-6cb5ff5c1a4737e846595bbb4e05719fR11. We can revisit this decision too, once we get some time to play with the welding

> You said this PR only touches preprocessing module, but there are commits that touch 1.generate-profiles/profiling_config.yaml and 1.generate-profiles/0.merge-single-cells.py. Do you want to revert them and move them to a different PR?

Nice catch. Can't sneak anything by! Since it is not doing any harm, how about I edit the note in the initial description, and then we can go ahead and merge?

**Comment by ErinWeisbart** [comment]
LGTM!

**Pull request merged**